lib/internal/loader/search.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const { URL } = require('url');
+const CJSmodule = require('module');
+const errors = require('internal/errors');
+const { resolve } = process.binding('module_wrap');
+const { Error, JSON: { stringify: JSONStringify } } = global;
+
+module.exports = (target, base) => {
+  target = `${target}`;
+  if (base === undefined) {
+    // We cannot search without a base.
+    throw new errors.Error('ERR_MISSING_MODULE', target);
+  }
+  base = `${base}`;
+  try {
+    return resolve(target, base);
+  } catch (e) {
+    e.stack; // cause V8 to generate stack before rethrow
+    let error = e;
+    try {
+      const questionedBase = new URL(base);
+      const tmpMod = new CJSmodule(questionedBase.pathname, null);
+      tmpMod.paths = CJSmodule._nodeModulePaths(
+        new URL('./', questionedBase).pathname);
+      const found = CJSmodule._resolveFilename(target, tmpMod);
+      error = new errors.Error(ERR_MODULE_RESOLUTION_DEPRECATED, target, base, found);
```

Comment:
Mind elaborating this error?

Reply:
ESM module resolution ignores `$NODE_PATH` or any module loading besides the regular relative `node_modules` resolution; this error tries to warn the user if the loading _would_ have found the module if ESM used the same resolution rules as NCJS.

Reply:
Ah, interesting. `deprecated` may be the wrong wording then, as it implies it would still work.

Reply:
changed to Legacy